src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,13 +35,32 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (isValidAddress(trimmedAddress)) {
```

Comment:
Probably should update the isValidAddress() to include new validation from the new parts, e.g. the block number.

Then you dont need a separate check for block below.